scripts/process/as_svg.R

```diff
@@ -0,0 +1,266 @@
+  
+  sfc_object <- .object[["geometry"]]
+  data_in <- data.frame(.object) %>% select(-geometry)
+  data_in$feature_id <- seq_len(nrow(data_in))
+  
+  # hmmm...with sf, each polygon of a multi-poly plots as a separate `d`. 
+  # Can get this w/ st_cast(sfc_object[j], "POLYGON")
+  # could test for this `"sfc_MULTIPOLYGON" %in% class(sfc_object[j])`
+  # use `feature_id` to track which feature each polygon came from
+  feature_ids <- c()
+  rendered <- svglite::xmlSVG(width = width, height = height, pointsize = pointsize, standalone = F, {
+    
+    set_svg_plot(xlim, ylim)
+    for (j in seq_len(length(sfc_object))){
+      for (poly in sf::st_cast(sfc_object[j], "POLYGON")){
+        plot(poly, ..., add = TRUE)
+        feature_ids <- c(feature_ids, j)
+      }
+    }
+  })
+  
+  
+  paths <- xml_child(rendered) %>% xml_children
+
+  # collapse the multi-d into a single one: 
+  data_out <- data.frame(d_raw = xml_attr(paths, 'd'), 
+                         feature_id = feature_ids, 
+                         stringsAsFactors = FALSE) %>% 
+    group_by(feature_id) %>% 
+    summarize(d = paste(d_raw, collapse = ' ')) %>% 
+    left_join(data_in) %>% select(-feature_id)
+  return(data_out) 
+}
+
+sf_to_text <- function(.object, xlim, ylim, ..., width, height, pointsize = 12){
+  stop('not implemented')
+}
+
+sf_to_circle <- function(.object, xlim, ylim, ..., width, height, pointsize = 12){
+  stop('not implemented')
+}
+
+#' turn spatial data into svg elements w/ attributes
+#' 
+#' @param viz a viz object, containing both `data` and `clip_box` as depends
+#' 
+#' @return a data.frame for use in an svg, containing the element name (`.value`), 
+#' `d`, and any other attributes added. See details
+#' @details 
+#' `<path>` assumes spatial data as inputs, which are then converted into `d` 
+#' attributes which describe the draw operations for svg paths
+#' the `clip_box` is used to figure out the x and y limits for plotting, which 
+#' set up the svg coordinate space that is needed to create the `d` attributes
+#' 
+#' Additional attributes can be added by using the optional `attributes` field 
+#' in the viz block. These must be character key value pairs, but can use mustache 
+#' text substitution for simple modifications (see `svg_sub_attributes`).
+#' 
+#' 
+#' in the future, maybe this function could also operate on non-spatial data
+process.as_svg_path <- function(viz){
+  
+  checkRequired(viz$depends, c('data', 'clip_box'))
+  
+  attributes <- viz[['attributes']]
+  svg <- as_svg_elements('path', viz)
+  
+  svg_path_out <- svg[['elements']] %>% 
+    svg_sub_attributes(attributes = attributes) %>% 
+    select_(.dots = names(attributes)) %>% 
+    do(sf_to_path(., xlim = svg$xlim, ylim = svg$ylim, width = svg$width, height = svg$height)) %>% 
+    mutate(.value = 'path') %>% select(.value, everything())
+    
+  
+  saveRDS(svg_path_out, viz[['location']])
+}
+
+#' turn data into svg elements w/ attributes
```

Comment:
For each of these `as_svg...` functions, it would help describing _why_ or in what instance each of these is used.